### PR TITLE
If package type is not found, do nothing when applying hints at the e…

### PR DESCRIPTION
**Pull request by dakaneye**
This should allow hints to provide content for go and binary content types

### CODE REVIEW

File: anchore_engine/analyzers/manager.py [hunk 1]
```diff
@@ -56,9 +57,8 @@ def _run_analyzer_modules(analyzer_report, configdir, imageId, unpackdir, output
                     )
                 else:
                     logger.debug(
-                        "command succeeded: cmd=%s stdout=%s stderr=%s".format(
-                            repr(cmdstr), repr(sout.strip()), repr(serr.strip())
-                        )
+                        "command succeeded: cmd=%s stdout=%s stderr=%s"
+                        % (repr(cmdstr), repr(sout.strip()), repr(serr.strip()))
```
Comment (alfredodeza) [inline anchore_engine/analyzers/manager.py:61]: if you are changing this already, it is safer to skip the `%` altogether and use the rest as arguments (see https://github.com/anchore/enterprise/issues/150 for reference):

```suggestion
                        "command succeeded: cmd=%s stdout=%s stderr=%s",
                        repr(cmdstr), repr(sout.strip()), repr(serr.strip())
```

Comment (dakaneye) [reply anchore_engine/analyzers/manager.py]: got it, fixed

**Review by Toure — APPROVED** [review]
LGTM

File: anchore_engine/analyzers/manager.py [hunk 2]
```diff
@@ -47,18 +47,18 @@ def _run_analyzer_modules(analyzer_report, configdir, imageId, unpackdir, output
                 serr = anchore_engine.utils.ensure_str(serr)
                 if rc != 0:
                     logger.error(
-                        "command failed: cmd=%s exitcode=%s stdout=%s stderr=%s".format(
-                            repr(cmdstr),
-                            repr(rc),
-                            repr(sout.strip()),
-                            repr(serr.strip()),
-                        )
+                        "command failed: cmd=%s exitcode=%s stdout=%s stderr=%s",
```
Comment (zhill) [inline anchore_engine/analyzers/manager.py:50]: Good fix. These needed updating anyway :)

File: anchore_engine/analyzers/manager.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def apply_hints(analyzer_report, unpackdir):
     # hints processing.
     for engine_entry in utils.content_hints(unpackdir=unpackdir):
         pkg_type = engine_entry.get("type")
-        if pkg_type:
+        if pkg_type and pkg_type in syft.modules_by_engine_type:
```
Comment (zhill) [inline anchore_engine/analyzers/manager.py:33]: I think an info message here would be good to indicate to the user that there are hints entry types not processed by the syft handler.

**Review by zhill — CHANGES_REQUESTED** [review]
Super minor change to add a log line. Otherwise looks good.

File: anchore_engine/analyzers/manager.py (hunk 3 above)
Comment (dakaneye) [reply anchore_engine/analyzers/manager.py:33]: @zhill that might be a bit confusing. The syft handler is not applying hints for certain package types, but they _are_ getting applied by the analyzers themselves

Comment (Toure) [reply anchore_engine/analyzers/manager.py:33]: @dakaneye I could move it to debug only?

**Comment by dakaneye** [comment]
Duplicated by #1008

**Pull request closed without merging**